src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -137,19 +175,27 @@ public partial class HubConnection
         /// <summary>
         /// Indicates the state of the <see cref="HubConnection"/> to the server.
         /// </summary>
-        public HubConnectionState State
-        {
-            get
-            {
-                // Copy reference for thread-safety
-                var connectionState = _connectionState;
-                if (connectionState == null || connectionState.Stopped)
-                {
-                    return HubConnectionState.Disconnected;
-                }
+        public HubConnectionState State { get; private set; }
```

Comment:
Small issue with this change is that the following code could now fail. We could decide it's not important enough, but I want to bring it up at least.
```csharp
if (connection.State == HubConnectionState.Connected || connection.State == HubConnectionState.Reconnecting)
{
    return something;
}
Assert.False(connection.State == HubConnectionState.Connected); // This could be hit now
```